xla/stream_executor/stream_executor_internal.h

```diff
@@ -336,11 +336,48 @@ class StreamExecutorInterface {
   // Performs linear search over alive GPU streams.
   virtual Stream* FindAllocatedStream(void* /*gpu_stream*/) { return nullptr; }
 
+  // The following methods access an internal log of some subset
+  // of arguments passed to other class methods.
+  // Used for testing/debugging purposes.
+
+  struct GemmCallTrace {
+    enum class GemmType {
+      kPlain = 0,
+      kStridedBatched = 1,
+      kBatched = 2,
+      kBlasLt = 3
+    };
+    GemmType op;
+    int flags;
+    uint64_t size1, size2;
+  };
+  // This may be expanded as necessary to trace other calls
+  using ApiTrace = std::variant<GemmCallTrace>;
+
+  // Retrieves and clears internal argument logs.
+  virtual absl::StatusOr<std::vector<ApiTrace> > ExtractApiTrace() {
+    return absl::UnimplementedError("Not implemented");
+  }
+
+  // Bit mask
+  enum class SE_ArgumentLogging: uint64_t {
+    kNone = 0,
+    kGemm = 1<<0,
+  };
+
+  // Sets the argument logging mode. Returns true if 'mode' is valid.
+  virtual bool SetArgumentLoggingMode(SE_ArgumentLogging mode) { return false; }
```

Comment:
Where's the intended eventual caller of this method?  I'm wondering if we need this all the way at the top level on StreamExecutorInterface.  Is this going to be ROCm-specific?  Can this just go on the rocm_blas class?

Reply:
The intended caller is here https://github.com/openxla/xla/pull/9593/files#diff-341de512548c6b2a2fa36f5208b4bc1274259307cebde81266e55cd36747b479R60

The initial use is ROCm-specific, but this is written in a way to be platform-independent, since this may be useful elsewhere and for other platforms.